From 09710d82c0a3469eadc32781721ac2336fdf915d Mon Sep 17 00:00:00 2001
Message-Id: <09710d82c0a3469eadc32781721ac2336fdf915d.1650800131.git.geliang.tang@suse.com>
From: Yucong Sun <fallentree@fb.com>
Date: Tue, 28 Sep 2021 11:42:21 -0700
Subject: [PATCH] bpftool: Avoid using "?: " in generated code
Git-commit: 09710d82c0a3469eadc32781721ac2336fdf915d
Patch-mainline: v5.16-rc1
References: bsn#127
Modified-by-SEL: No


"?:" is a GNU C extension, some environment has warning flags for its
use, or even prohibit it directly.  This patch avoid triggering these
problems by simply expand it to its full form, no functionality change.

Signed-off-by: Yucong Sun <fallentree@fb.com>
Signed-off-by: Andrii Nakryiko <andrii@kernel.org>
Link: https://lore.kernel.org/bpf/20210928184221.1545079-1-fallentree@fb.com
Signed-off-by: Geliang Tang <geliang.tang@suse.com>
---
 tools/bpf/bpftool/gen.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/tools/bpf/bpftool/gen.c b/tools/bpf/bpftool/gen.c
index e3ec47a6a612..cc835859465b 100644
--- a/tools/bpf/bpftool/gen.c
+++ b/tools/bpf/bpftool/gen.c
@@ -803,7 +803,10 @@ static int do_skeleton(int argc, char **argv)
 			}						    \n\
 									    \n\
 			err = %1$s__create_skeleton(obj);		    \n\
-			err = err ?: bpf_object__open_skeleton(obj->skeleton, opts);\n\
+			if (err)					    \n\
+				goto err_out;				    \n\
+									    \n\
+			err = bpf_object__open_skeleton(obj->skeleton, opts);\n\
 			if (err)					    \n\
 				goto err_out;				    \n\
 									    \n\
-- 
2.26.2

